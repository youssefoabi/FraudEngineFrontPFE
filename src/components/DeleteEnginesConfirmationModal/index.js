import { bool, func } from 'prop-types';
import { contains, isEmpty, isNil, join, length, pathOr, pickBy, pipe, pluck, values } from 'ramda';
import { Typography } from '@material-ui/core';
import { useDispatch, useSelector } from 'react-redux';
import React, { useCallback } from 'react';

import { DELETE_ENGINES_REQUEST } from '../EnginesDataGrid/actions';
import ConfirmationModal from '../ConfirmationModal';

export default function DeleteEnginesConfirmationModal({ open, toggle }) {
  const dispatch = useDispatch();

  const deleteEngines = useCallback(() => dispatch({ type: DELETE_ENGINES_REQUEST }), [dispatch]);

  const selected = useSelector(pathOr([], ['navigation', 'engines', 'selected']));
  const namesSelector = pipe(
    pathOr([], ['entities', 'engines', 'byId']),
    pickBy((value, key) => contains(+key, selected)),
    values,
    pluck('name'),
    join(', '),
  );
  const names = useSelector(namesSelector);

  if (isNil(names) || isEmpty(names)) return null;

  const deleteMoreThanOneEngine = !isNil(selected) && length(selected) > 1;
  const withS = deleteMoreThanOneEngine ? 's' : '';
  const withU = deleteMoreThanOneEngine ? 'es' : 'u';

  return (
    <ConfirmationModal
      isOpen={open}
      title={`Suppression d${withU} moteur${withS}`}
      toggle={toggle}
      confirm={deleteEngines}
    >
      <Typography>{`Confirmez-vous la suppression d${withU} moteur${withS} suivant${withS} : ${names} ?`}</Typography>
    </ConfirmationModal>
  );
}

DeleteEnginesConfirmationModal.propTypes = {
  open: bool,
  toggle: func,
};

DeleteEnginesConfirmationModal.defaultProps = {
  open: false,
  toggle: Function.prototype,
};
