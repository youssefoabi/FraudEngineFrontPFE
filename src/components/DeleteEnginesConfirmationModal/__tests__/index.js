import { itRendersCorrectly } from '@cdiscount/jest-utils';
import React from 'react';

import DeleteEnginesConfirmationModal from '../';

jest.mock('react-redux', () => ({
  useDispatch: () => ({}),
  useSelector: jest
    .fn()
    .mockReturnValueOnce([])
    .mockReturnValueOnce([])
    .mockReturnValueOnce([1])
    .mockReturnValueOnce('engine_1')
    .mockReturnValueOnce([1, 2])
    .mockReturnValueOnce('engine_1, engine_2'),
}));

describe(DeleteEnginesConfirmationModal.name, () => {
  describe('without selected engines', () => {
    itRendersCorrectly(<DeleteEnginesConfirmationModal />);
  });

  describe('with select one engine', () => {
    itRendersCorrectly(<DeleteEnginesConfirmationModal />);
  });

  describe('with select more than one engine', () => {
    itRendersCorrectly(<DeleteEnginesConfirmationModal />);
  });
});
