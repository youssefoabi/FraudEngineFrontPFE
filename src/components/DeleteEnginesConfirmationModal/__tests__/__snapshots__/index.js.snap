// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DeleteEnginesConfirmationModal with select more than one engine renders correctly 1`] = `
<ConfirmationModal
  confirm={[Function]}
  isOpen={false}
  title="Suppression des moteurs"
  toggle={[Function]}
>
  <WithStyles(ForwardRef(Typography))>
    Confirmez-vous la suppression des moteurs suivants : engine_1, engine_2 ?
  </WithStyles(ForwardRef(Typography))>
</ConfirmationModal>
`;

exports[`DeleteEnginesConfirmationModal with select one engine renders correctly 1`] = `
<ConfirmationModal
  confirm={[Function]}
  isOpen={false}
  title="Suppression du moteur"
  toggle={[Function]}
>
  <WithStyles(ForwardRef(Typography))>
    Confirmez-vous la suppression du moteur suivant : engine_1 ?
  </WithStyles(ForwardRef(Typography))>
</ConfirmationModal>
`;

exports[`DeleteEnginesConfirmationModal without selected engines renders correctly 1`] = `""`;
